src/Fixer/LanguageConstruct/FunctionToConstantFixer.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\LanguageConstruct;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ */
+final class FunctionToConstantFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    /**
+     * @var array<string, string>
+     */
+    private static $defaultConfiguration = array(
+        'phpversion' => 'PHP_VERSION',
+        'php_sapi_name' => 'PHP_SAPI',
+        'pi' => 'M_PI',
+    );
+
+    /**
+     * @var array<string, string>
+     */
+    private $configuration;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        if (null === $configuration) {
+            $this->configuration = self::$defaultConfiguration;
+
+            return;
+        }
+
+        $this->configuration = array();
+        foreach ($configuration as $key => $value) {
+            if (!is_string($key) || !array_key_exists($key, self::$defaultConfiguration)) {
+                throw new InvalidFixerConfigurationException($this->getName(), sprintf('Configuration key must be any of "%s".', implode('", "', array_keys(self::$defaultConfiguration))));
+            }
+
+            if (!is_bool($value)) {
+                throw new InvalidFixerConfigurationException(
+                    $this->getName(),
+                    sprintf(
+                        'Configuration value must be an bool, got "%s" for "%s".',
+                        is_object($value) ? get_class($value) : (gettype($value).'#'.$value),
+                        $key
+                    )
+                );
+            }
+
+            if ($value) {
```

Comment:
What we can't do anymore with #2500 is passing an array with keys that don't match option names (numeric keys or undefined options).

Here we had this: `['foo' => 'FOO', 'bar' => 'BAR']`, which can be seen as _"the configuration is a mapping of functions with constants"_. Strictly speaking, it _is_ compatible with #2500, assuming the option names are `foo` and `bar`.

But if you want to support replacing more functions, you have to create a new option for each. And if you want to add an option to tweak the behavior instead of extending supported functions, you end up mixing "mapping" options with "behavior" options. This is why I suggested passing the mapping in a dedicated `replacements` option (or any better name).

Thinking again, I think boolean options here are not the best solution because you may still mix "mapping" and "behavior" options, and you lose the ability to choose the constants.